### PR TITLE
feat: 🔥 Add random super hero functions

### PR DESCRIPTION
**Pull request by theryansmee**
Add randSuperHero. Add randSuperHeroName

_Also, added last of missing char to full-name regex_

✅ Closes: #134 

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/falso/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: 134

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by theryansmee** [comment]
I wasn't sure if the category for `randSuperhero` should be `entity`? If so, which category should `randSuperheroName` be under?

**Comment by NetanelBasal** [comment]
randSuperhero - entity, randSuperheroName - comics?

File: packages/falso/src/lib/superhero.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * @example
+ *
+ * randSuperhero({ company: 'Marvel' })
+ *
+ * @example
+ *
+ * randSuperheroName({ length: 10 })
+ *
+ */
+export function randSuperheroName<Options extends SuperheroOptions>(
```
[the remark this answers is not in the document]
Comment (theryansmee) [reply packages/falso/src/lib/superhero.ts:75]: Ok I will do. Should I also create a separate file for shared interfaces and types? Or just keep them in the entity file?

[the remark this answers is not in the document]
Comment (NetanelBasal) [reply packages/falso/src/lib/superhero.ts:75]: only for the rand functions

File: packages/falso/src/lib/superhero.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+ *
+ * @example
+ *
+ * randSuperhero({ length: 10 })
+ *
+ */
+export function randSuperhero<Options extends SuperheroOptions>(
+  options?: Options
+) {
+  const filteredData: SuperheroEntity[] = data
```
Comment (NetanelBasal) [inline packages/falso/src/lib/superhero.ts:43]: Why do you filter if the user doesn't pass a company? Use the whole data directly.

File: packages/falso/src/lib/superhero.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import { randUuid } from '@ngneat/falso';
+
+export type ComicBookCompany = 'Marvel' | 'DC';
+
+export interface SuperheroOptions extends FakeOptions {
+  company?: ComicBookCompany;
+}
+
+export interface Superhero {
+  id?: string;
```
Comment (NetanelBasal) [inline packages/falso/src/lib/superhero.ts:12]: Why `id?`. You can remove it

File: packages/falso/src/lib/superhero-name.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ * @example
+ *
+ * randSuperheroName({ length: 10 })
+ *
+ */
+export function randSuperheroName<Options extends SuperheroOptions>(
+  options?: Options
+) {
+  const factory: () => string = () => {
+    const heroes = data
```
Comment (NetanelBasal) [inline packages/falso/src/lib/superhero-name.ts:28]: Why do you filter if the user doesn't pass a company? Use the whole data directly.

File: packages/falso/src/tests/superhero-name.spec.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  describe('company is passed', () => {
+    let company: ComicBookCompany;
+
+    describe('DC is passed', () => {
+      beforeEach(() => {
+        company = 'DC';
+      });
+
+      it('should return a random super superhero name from DC', () => {
+        const result: string = randSuperheroName({ company });
```
Comment (NetanelBasal) [inline packages/falso/src/tests/superhero-name.spec.ts:23]:
```suggestion
        const result = randSuperheroName({ company });
```

File: packages/falso/src/tests/superhero-name.spec.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        expect(allNames).toContain(result);
+        expect(result).toEqual(expect.any(String));
+      });
+    });
+  });
+
+  describe('length is passed', () => {
+    describe('length is 2', () => {
+      it('should return a string with 2 random superhero names', () => {
+        const result: string[] = randSuperheroName({ length: 2 });
```
Comment (NetanelBasal) [inline packages/falso/src/tests/superhero-name.spec.ts:53]:
```suggestion
        const result = randSuperheroName({ length: 2 });
```

File: packages/falso/src/tests/superhero-name.spec.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+      });
+    });
+
+    describe('Marvel is passed', () => {
+      beforeEach(() => {
+        company = 'Marvel';
+      });
+
+      it('should return a random super superhero name from Marvel', () => {
+        const result: string = randSuperheroName({ company });
```
Comment (NetanelBasal) [inline packages/falso/src/tests/superhero-name.spec.ts:39]:
```suggestion
        const result = randSuperheroName({ company });
```

File: packages/falso/src/tests/superhero-name.spec.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+      it('should return a string with 2 random superhero names', () => {
+        const result: string[] = randSuperheroName({ length: 2 });
+
+        expect(result.length).toEqual(2);
+      });
+    });
+
+    describe('length is 5', () => {
+      it('should return a string with 5 random superhero names', () => {
+        const result: string[] = randSuperheroName({ length: 5 });
```
Comment (NetanelBasal) [inline packages/falso/src/tests/superhero-name.spec.ts:61]: It should be automatically inferred, there is no need for this.

File: packages/falso/src/tests/superhero.spec.ts [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+      });
+    });
+
+    describe('Marvel is passed', () => {
+      beforeEach(() => {
+        company = 'Marvel';
+      });
+
+      it('should return a random super superhero entity from Marvel', () => {
+        const result: SuperheroEntity = randSuperhero({ company });
```
Comment (NetanelBasal) [inline packages/falso/src/tests/superhero.spec.ts:45]: same in this file

File: packages/falso/src/lib/superhero.ts (hunk 3 above)
Comment (theryansmee) [reply packages/falso/src/lib/superhero.ts]: I didnt mean to leave this one in 😊

File: packages/falso/src/lib/superhero-name.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+ * @example
+ *
+ * randSuperhero({ company: 'Marvel' })
+ *
+ * @example
+ *
+ * randSuperheroName({ length: 10 })
+ *
+ */
+export function randSuperheroName<Options extends SuperheroOptions>(
```
Comment (NetanelBasal) [inline packages/falso/src/lib/superhero-name.ts:24]: Please pull main and update to `Options extends SuperheroOptions = never`

File: packages/falso/src/lib/superhero.ts [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+ * @example
+ *
+ * randSuperhero({ company: 'Marvel' })
+ *
+ * @example
+ *
+ * randSuperhero({ length: 10 })
+ *
+ */
+export function randSuperhero<Options extends SuperheroOptions>(
```
Comment (NetanelBasal) [inline packages/falso/src/lib/superhero.ts:39]: Please pull main and update to `Options extends SuperheroOptions = never`

**Comment by NetanelBasal** [comment]
CI is failing

**Comment by theryansmee** [comment]
@NetanelBasal - I'm fixing it now and adding a few tests to ensure it doesnt happen again. Sorry

**Pull request merged**